### PR TITLE
refactor to break BaseResourceObject up

### PR DESCRIPTION
**Pull request by scytacki**
move resource objects to the models folder
pull out the static methods of BaseResourceObject into a resource manager module
move S3ResourceObject and IotResourceObject into their own files
add additional tests to cover error messages,  resource-manager has 98% coverage now

### CODE REVIEW

File: functions/src/models/iot-resource-object.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import {
+  AuthClaims
+} from "../firestore-types";
+import {BaseResourceObject} from "./base-resource-object";
+import {
+  Credentials, Config, IotResource
+} from "../resource-types";
+
+export class IotResourceObject extends BaseResourceObject {
+  // TODO: add iot specific resource members
```
Comment (dougmartin) [inline functions/src/models/iot-resource-object.ts:10]: Was this a TODO for this PR or for later?

Comment (dougmartin) [reply functions/src/models/iot-resource-object.ts:10]: Same question with the other TODOs

**Review by dougmartin — APPROVED** [review]
Looks good :+1:

File: functions/src/models/iot-resource-object.ts (hunk 1 above)
Comment (scytacki) [reply functions/src/models/iot-resource-object.ts:10]: Thanks for asking. These are for later. They were actually there before the refactoring. We never actually used this IotResource in the DataFlow project. But I'm about to add a AthenaWorkgroup Resource which hopefully we will use, so the IotResource served as a good place holder to help with the separation.

**Review by pjanik — APPROVED** [review]
Looks good to me 👍

**Pull request merged**